File: ocaml/leap/leap.ml
let leap_year year =
  let is_year_div_by x =
    Int.rem year x = 0
  in
  match () with 
  | _ when is_year_div_by 400 -> true
  | _ when is_year_div_by 100 -> false
  | _ -> is_year_div_by 4

File: solutions/python/leap/1/leap.py
"""
Module with a function to calculate leap years
"""

def leap_year(year):
    """
    Calculate if year is a leap year
    year : int
    return : bool
    """
    return year % 400 == 0 \
        or (not year % 100 == 0 and year % 4 == 0)
